hyperdash_cli/cli.py

```diff
@@ -284,6 +241,62 @@ def stderr_loop():
     wrapped()
 
 
+def pipe(args):
+    @_monitor(args.name, api_key_getter=None, capture_io=True, api_name=API_NAME_CLI_PIPE)
+    def wrapped():
+        # Read STDIN and write it to STDOUT so it shows up in the terminal and is
+        # captured by the monitor decorator
+        _connect_streams(sys.stdin, sys.stdout)
+    wrapped()
+
+
+# Reads a pipe one character at a time, yielding
+# buffers everytime it encounters whitespace. This
+# allows us read the pipe as fast as possible.
+#
+# We yield everytime we encounter whitespace instead
+# of on every byte because yielding every byte individually
+# would break the UTF-8 decoding of multi-byte characters.
+#
+# Before this we were using readline() which works in most
+# cases, but breaks for scripts that use loading bars
+# like tqdm which do not output a \n everytime they
+# update. Using read() doesn't work either because there
+# is no guarantee of when that will be flushed so
+# terminal updates can be delayed.
+def _gen_tokens_from_stream(stream):
+    buf = []
+    while True:
+        # read one byte        
+        if PY2:
+            b = stream.read(1)
+        else:
+            b = stream.buffer.read(1)
```

Comment:
Might need a different approach here

> Note To write or read binary data from/to the standard streams, use the underlying binary buffer object. For example, to write bytes to stdout, use sys.stdout.buffer.write(b'abc').
However, if you are writing a library (and do not control in which context its code will be executed), be aware that the standard streams may be replaced with file-like objects like io.StringIO which do not support the buffer attribute.

https://docs.python.org/3.5/library/sys.html#sys.stdin

Reply:
I think it will actually be fine because this is not a library use-case, the pipe command is basically a program that executes on its own. Its not embedded

Reply:
also if someone replaces sys.stdin with an IOBuffer or something like that all our code is screwed anyways